File: Sem3/Advanced Methods of Programming/MyInterpreter - JavaFX Graphical Application/project/src/domain/expression/ConstantExpression.java
package domain.expression;


import domain.adt.MyDictionaryInterface;


public class ConstantExpression extends Expression
{
    private int number;


    public ConstantExpression(int number)
    {
        this.number = number;
    }


    @Override
    public int eval(MyDictionaryInterface<String, Integer> symbolTable,
                    MyDictionaryInterface<Integer, Integer> heapTable) throws ExpressionException
    {
        return number;
    }


    @Override
    public String toString()
    {
        return Integer.toString(number);
    }
}

File: Sem3/Advanced Methods of Programming/MyInterpreter - CSharp Console Application/MyInterpreter CSharp/domain/expression/ConstantExpression.cs
using MyInterpreter_CSharp.domain.adt;

namespace MyInterpreter_CSharp.domain.expression
{
    public class ConstantExpression : Expression
    {
        private int _number;

        public ConstantExpression(int number)
        {
            this._number = number;
        }

        public override int Eval(IMyDictionary<string, int> symbolTable)
        {
            return _number;
        }

        public override string ToString()
        {
            return _number.ToString();
        }
    }
}